WordPress/Classes/ViewRelated/Post/Scheduling/CalendarCollectionView.swift

```diff
@@ -75,34 +108,64 @@ extension CalendarDataSource: JTACMonthViewDelegate {
     }
 
     func calendar(_ calendar: JTACMonthView, didSelectDate date: Date, cell: JTACDayCell?, cellState: CellState, indexPath: IndexPath) {
+        if style == .year, let firstDate = firstDate {
+            calendar.selectDates(from: firstDate,
+                                 to: date,
+                                 triggerSelectionDelegate: false,
+                                 keepSelectionIfMultiSelectionAllowed: true)
+            didSelect?(firstDate, date)
+        } else {
+            firstDate = date
+            didSelect?(date, nil)
+        }
+
         configure(cell: cell, with: cellState)
-        didSelect?(date)
     }
 
     func calendar(_ calendar: JTACMonthView, didDeselectDate date: Date, cell: JTACDayCell?, cellState: CellState, indexPath: IndexPath) {
         configure(cell: cell, with: cellState)
     }
 
-    private func configure(cell: JTACDayCell?, with state: CellState) {
-        let cell = cell as? DateCell
-        cell?.configure(with: state)
+    func calendar(_ calendar: JTACMonthView, shouldSelectDate date: Date, cell: JTACDayCell?, cellState: CellState, indexPath: IndexPath) -> Bool {
+        if style == .year, calendar.selectedDates.count > 1 && cellState.selectionType != .programatic || firstDate != nil && !calendar.selectedDates.isEmpty && date < calendar.selectedDates[0] {
+            firstDate = nil
+            let retval = !calendar.selectedDates.contains(date)
+            didDeselectAllDates?()
+            calendar.deselectAllDates()
```

Comment:
Minor idea, since the method is `did` deselect shouldn't this be called after the deselection happens?
```suggestion
            calendar.deselectAllDates()
            didDeselectAllDates?()
```

Reply:
Same as above.